qualang_tools/analysis/discriminator.py

```diff
@@ -0,0 +1,131 @@
+import numpy as np
+from matplotlib import pyplot as plt
+from scipy.optimize import minimize
+
+
+def _false_detections(threshold, Ig, Ie):
+    if np.mean(Ig) < np.mean(Ie):
+        false_detections_var = np.sum(Ig > threshold) + np.sum(Ie < threshold)
+    else:
+        false_detections_var = np.sum(Ig < threshold) + np.sum(Ie > threshold)
+    return false_detections_var
+
+
+def two_state_discriminator(Ig, Qg, Ie, Qe, b_print=True, b_plot=True):
+    """
+    Given two blobs in the IQ plane representing two states, finds the optimal threshold to discriminate between them
+    and calculates the fidelity. Also returns the angle in which the data needs to be rotated in order to have all the
+    information in the `I` (`X`) axis.
+
+    .. note::
+        This function assumes that there are only two blobs in the IQ plane representing two states (ground and excited)
+        Unexpected output will be returned in other cases.
+
+
+    :param float Ig: A vector containing the `I` quadrature of data points in the ground state
+    :param float Qg: A vector containing the `Q` quadrature of data points in the ground state
+    :param float Ie: A vector containing the `I` quadrature of data points in the excited state
+    :param float Qe: A vector containing the `Q` quadrature of data points in the excited state
+    :param bool b_print: When true (default), prints the results to the console.
+    :param bool b_plot: When true (default), plots the results in a new figure.
+    :returns: A tuple of (angle, threshold, b_excited_is_larger, fidelity, gg, ge, eg, ee).
```

Comment:
b_excited_is_larger is not returned anymore

Reply:
nice catch :)